test/org/sagebionetworks/bridge/models/schedules/CriteriaScheduleStrategyTest.java

```diff
@@ -0,0 +1,416 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.sagebionetworks.bridge.TestConstants.TEST_STUDY_IDENTIFIER;
+
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.sagebionetworks.bridge.TestConstants;
+import org.sagebionetworks.bridge.dynamodb.DynamoSchedulePlan;
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.json.BridgeObjectMapper;
+import org.sagebionetworks.bridge.models.ClientInfo;
+import org.sagebionetworks.bridge.models.accounts.User;
+import org.sagebionetworks.bridge.validators.SchedulePlanValidator;
+import org.sagebionetworks.bridge.validators.Validate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.google.common.collect.Sets;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+public class CriteriaScheduleStrategyTest {
+    
+    private static final Schedule SCHEDULE_FOR_STRATEGY_WITH_APP_VERSIONS = makeValidSchedule(
+            "Strategy With App Versions");
+    private static final Schedule SCHEDULE_FOR_STRATEGY_WITH_ONE_REQUIRED_DATA_GROUP = makeValidSchedule(
+            "Strategy With One Required Data Group");
+    private static final Schedule SCHEDULE_FOR_STRATEGY_WITH_REQUIRED_DATA_GROUPS = makeValidSchedule(
+            "Strategy With Required Data Groups");
+    private static final Schedule SCHEDULE_FOR_STRATEGY_WITH_ONE_PROHIBITED_DATA_GROUP = makeValidSchedule(
+            "Strategy With One Prohibited Data Group");
+    private static final Schedule SCHEDULE_FOR_STRATEGY_WITH_PROHIBITED_DATA_GROUPS = makeValidSchedule(
+            "Strategy With One Prohibited Data Groups");
+    private static final Schedule SCHEDULE_FOR_STRATEGY_NO_CRITERIA = makeValidSchedule("Strategy No Criteria");
+    private static final Schedule SCHEDULE_FOR_STRATEGY_WITH_ALL_REQUIREMENTS = makeValidSchedule(
+            "Strategy with all requirements");
+    
+    private CriteriaScheduleStrategy strategy;
+    private static final SchedulePlan PLAN = new DynamoSchedulePlan();
+    private static final SchedulePlanValidator VALIDATOR = new SchedulePlanValidator(Sets.newHashSet(),
+            Sets.newHashSet(TestConstants.TEST_3_ACTIVITY.getTask().getIdentifier()));;
+    
+    @Test
+    public void hashCodeEquals() {
+        EqualsVerifier.forClass(CriteriaScheduleStrategy.class).allFieldsShouldBeUsed().verify();
+    }
+    
+    @Before
+    public void before() {
+        strategy = new CriteriaScheduleStrategy();
+
+        PLAN.setLabel("Schedule plan label");
+        PLAN.setStudyKey(TEST_STUDY_IDENTIFIER);
+        PLAN.setStrategy(strategy);
```

Comment:
Minor: PLAN.set* could be moved to a static initializer so we don't have to make these redundant calls with each unit test.